part2/phonebook/src/App.js

```diff
@@ -0,0 +1,80 @@
+import { useState } from 'react'
+
+import PersonList from './components/PersonList.js'
+import PersonForm from './components/PersonForm.js'
+import Filter from './components/Filter.js'
+
+const App = () => {
+  const [persons, setPersons] = useState([
+    { name: 'Arto Hellas', number: '0412 345 678', id: 1},
+    { name: 'Ada Lovelace', number: '39-44-5323523', id: 2 },
+    { name: 'Dan Abramov', number: '12-43-234345', id: 3 },
+    { name: 'Mary Poppendieck', number: '39-23-6423122', id: 4 }
+  ]) 
+  const [newName, setNewName] = useState('')
+  const [newNumber, setNewNumber] = useState('')
+  const [searchName, setSearchName] = useState('')
+  const [filter, setFilter] = useState(false)
+
+  const personToShow = filter
+    ? persons.filter(person => person.name.toLowerCase().includes(searchName.toLowerCase()))
+    : persons
+  
+  const addName = (event) => {
+    event.preventDefault()
+    console.log('button clicked', event.target)
+
+    console.log(checkName)
+
+    if (checkName === true) {
+      alert(`${newName} is already added to phonebook` ) 
+    }
+    else {
+      setPersons(persons.concat({name : newName, number: newNumber, id: persons.length+1}))
+      setNewName('')
+      setNewNumber('')
+    }
+  }
+
+  const checkName = persons.some(person => {
+    if (person.name === newName) {
+      return true
+    }
+    return false
+  })
```

Comment:
declare variable before using it, javascript lets u do this way too but best practice is to declare a variable first